File: Implementierung/Backend/src/main/java/edu/kit/informatik/dto/mapper/TableMapper.java
package edu.kit.informatik.dto.mapper;

import edu.kit.informatik.dto.userdata.rooms.TableDto;
import edu.kit.informatik.model.userdata.rooms.Table;

import java.util.List;

public class TableMapper implements IModelDtoMapper<Table, TableDto> {

    @Override
    public TableDto modelToDto(Table table) {
        return null;
    }

    @Override
    public List<TableDto> modelToDto(List<Table> tables) {
        return null;
    }

    @Override
    public Table dtoToModel(TableDto tableDto) {
        return null;
    }

    @Override
    public List<Table> dtoToModel(List<TableDto> tableDtos) {
        return null;
    }
}

File: Implementierung/frontend/paschi/src/dto/mapper/rooms/TableMapper.ts
import {IModelDtoMapper} from "@/dto/mapper/IModelDtoMapper";
import {Table} from "@/model/userdata/rooms/Table";
import {TableDto} from "@/dto/userdata/rooms/TableDto";
import {PositionMapper} from "@/dto/mapper/rooms/PositionMapper";
import {UserController} from "@/controller/UserController";
import {useRoomObjectStore} from "@/store/RoomObjectStore";

export class TableMapper implements IModelDtoMapper<Table, TableDto> {

  private static mapper: TableMapper = new TableMapper();
  private positionMapper = PositionMapper.getMapper();

  private constructor() {
  }

  static getMapper(): TableMapper {
    return TableMapper.mapper;
  }

  modelToDto(table: Table): TableDto {
    return new TableDto(
      table.getId,
      table.user.getId,
      table.createdAt,
      table.updatedAt,
      this.positionMapper.modelToDto(table.position),
      table.dimensions.length,
      table.dimensions.width
    );
  }

  async dtoToModel(tableDto: TableDto): Promise<Table> {
    const userController = UserController.getUserController();

    const position = await this.positionMapper.dtoToModel(tableDto.position)

    let table = useRoomObjectStore().getTableByTimeCreatedAndPosition(tableDto.createdAt, position);
    if (table == undefined) {
      table = new Table(
        tableDto.id,
        0,
        userController.getUser(),
        position,
        tableDto.length,
        tableDto.width
      );
      table.updatedAt = tableDto.updatedAt;
      table.createdAt = tableDto.createdAt;
      useRoomObjectStore().addChair(table);
    }

    table.setId = tableDto.id;
    table.position = position;
    table.dimensions.length = tableDto.length;
    table.dimensions.width = tableDto.width;

    return table;
  }
}
